### PR TITLE
Parafac2

### PR DESCRIPTION
**Pull request by andrewram4287**

### CODE REVIEW

**Comment by andrewram4287** [comment]
@aarmey I thought parafac2 would be able to handle different sizes (i.e. NaN's) for the cell dimension?

File: gmm/figures/figure0.py [hunk 1]
```diff
@@ -46,4 +51,33 @@ def makeFigure():
...
+    for i in range(len(experimentCellCount)):
+        cells = np.arange(1, int(experimentCellCount[i]) + 1)
+        cellCount = np.append(cellCount,cells)
+        
+    PopAlignDF["Cell"] = cellCount
+    
+    PopAlignDF["Cell Type"] = cell_types.values
+    PopAlignXA = PopAlignDF.set_index(["Cell", "Drug"]).to_xarray()
+    CellTypeXA = PopAlignXA["Cell Type"]
+    PopAlignXA = PopAlignXA[["C21ORF58","PCNT", "S100B","AC145212.1"]].to_array(dim="Gene")
```
Comment (andrewram4287) [inline gmm/figures/figure0.py:79]: Only using a few genes at first to make sure parafac2 works

**Comment by aarmey** [comment]
Yes, but you don't fill in with NaNs—you pass in a list of 2D matrices of different sizes. (Weird, I know.)

**Comment by aarmey** [comment]
I *think* the data import I just pushed should work. (See my comment about a potential caveat we'll want to confirm.)

**Comment by aarmey** [comment]
Merging before I duplicate the repo.

**Comment by andrewram4287** [comment]
@aarmey I was having difficulty yesterday trying to figure out different methods for the import, thanks for helping out!

**Pull request merged**